community/modules/compute/schedmd-slurm-gcp-v5-node-group/source_image_logic.tf

```diff
@@ -15,12 +15,53 @@
  */
 
 locals {
-  source_image         = lookup(var.instance_image, "name", "")
-  source_image_family  = lookup(var.instance_image, "family", "")
-  source_image_project = lookup(var.instance_image, "project", "")
-  source_image_project_normalized = (
-    local.source_image != "" || length(regexall("/", local.source_image_project)) > 0
-    ? local.source_image_project
-    : "projects/${local.source_image_project}/global/images/family"
+  # Currently supported images and projects
+  known_project_families = {
+    schedmd-slurm-public = [
+      "slurm-gcp-5-7-debian-11",
+      "slurm-gcp-5-7-hpc-rocky-linux-8",
+      "slurm-gcp-5-7-ubuntu-2004-lts",
+      "slurm-gcp-5-7-ubuntu-2204-lts-arm64",
+      "slurm-gcp-5-7-hpc-centos-7-k80",
+      "slurm-gcp-5-7-hpc-centos-7"
+    ]
+  }
+
+  # This approach to "hacking" the project name allows a chain of Terraform
+  # calls to set the instance source_image (boot disk) with a "relative 
+  # resource name" that passes muster with VPC Service Control rules
+  #
+  # https://github.com/terraform-google-modules/terraform-google-vm/blob/735bd415fc5f034d46aa0de7922e8fada2327c0c/modules/instance_template/main.tf#L28
+  # https://cloud.google.com/apis/design/resource_names#relative_resource_name
+  source_image_project_normalized = (can(var.instance_image.family) ?
+    "projects/${data.google_compute_image.slurm.project}/global/images/family" :
+    "projects/${data.google_compute_image.slurm.project}/global/images"
   )
+  source_image_family = can(var.instance_image.family) ? data.google_compute_image.slurm.family : ""
+  source_image        = can(var.instance_image.name) ? data.google_compute_image.slurm.name : ""
+}
+
+data "google_compute_image" "slurm" {
+  family  = try(var.instance_image.family, null)
+  name    = try(var.instance_image.name, null)
+  project = var.instance_image.project
+
+  lifecycle {
+    postcondition {
+      condition     = var.instance_image_custom || contains(keys(local.known_project_families), self.project)
+      error_message = <<-EOD
+      '${self.project}' is not a known project with compatible Slurm image families. Use the 'instance_image_custom' flag to deploy custom images. See: https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/vm-images.md#slurm-on-gcp.
```

Comment:
In interest of closing loop more quickly, I suggest that you modify this message to read:

```
      Images in project ${self.project} are not published by SchedMD. Images must be
      built by compatible releases of the Terraform and Packer modules following the
      guidance at https://goo.gle/hpc-slurm-images. Set var.instance_image_custom to
      true to silence this error and acknowledge that you are using a compatible image.     
```